swift-sdk/Internal/in-app/InAppManager+Functions.swift

```diff
@@ -59,7 +63,11 @@ struct MessagesProcessor {
         
         ITBDebug("isOkToShowNow")
         
-        if inAppDelegate.onNew(message: message) == .show {
+        let returnValue = inAppDelegate.onNew(message: message)
+        if message.isJsonOnly {
+            return .skipAndConsume(message)
+        }
+        if returnValue == .show {
```

Comment:
Maybe we don't need to hold the inAppDelegate onNew value in a var.

```suggestion
        if message.isJsonOnly {
            return .skipAndConsume(message)
        }
        if inAppDelegate.onNew(message: message) == .show {
```

Reply:
well we have to call onNew on a json only message before returning so thats why we call and store the return value and then ignore it for json only